src/render-delaying-services.js

```diff
@@ -56,11 +76,25 @@ const LOAD_TIMEOUT = 3000;
  *     as the detected render delaying services
  */
 export function waitForServices(win) {
-  const promises = includedServices(win).map(service => {
+  const promises = includedServices(win).map(serviceId => {
+
+    const serviceReadyPromise = getServicePromise(
+        win,
+        serviceId
+    ).then(service => {
+      if (service && service.whenReady) {
```

Comment:
I think you should actually typecast it to `RenderDelayingService` interface before you check for existence of `whenReady`. I recommend to have some sort of function right next to the interface definition:

```
/**
 * @param {!Object} service
 * @return {?RenderDelayingService}
 */
export function ifRenderDelayingService(service) {
  const maybeRenderDelayingService =
    /** @type {!RenderDelayingService}*/ (service);
  if (typeof maybeRenderDelayingService.whenReady == 'function') {
    return maybeRenderDelayingService;
  }
  return null;
}
```

/cc @erwinmombay